### PR TITLE
#273 bug fix contacts black screen

### LINKED ISSUE #273: BUG-273: Loading "Invite By Contact List" causes black screen when user has many contacts
### Related Issue(s): 
#117

### Relevant File(s):
[SearchContactsToSupervise.java](https://github.com/Landry333/Big-Owl/blob/79cd8e8b456c7d44bca68c4da307527bc3456630/app/src/main/java/com/example/bigowlapp/activity/SearchContactsToSupervise.java)

### Describe the bug:
When the user has many contacts, when going to the page to invite a user using their contact list, going to that page can result in a long black screen, as if the app has froze.

### To Reproduce:
1. Add many contacts to the Android device (100 or greater)
2. Run the app and login as any user if not already logged in
3. Click "Add Users"
4. Click "Invite from Contact List"

The screen will be frozen black but eventually load the contacts.

### Expected behavior
We should get to the contacts page and see our contacts. At worst we should get to a normal page and the contacts would be loading.

### Additional context: <optional>

https://stackoverflow.com/questions/26863223/android-black-screen-when-get-phone-contacts

Loading contacts is heavy, so we should probably consider loading them asynchronously, currently, this is not the case.  Infinite scrolling and pagination might also need to be considered, as a long load time, while an improvement from a black screen, is also not great.

### PR DESCRIPTION
**Pull request by refatK**
### Related Issues
Closes #273

### Description
Contacts require a database call to get in Android, so I followed Along with Android's explanation of how to get a Contacts list and used a `LoaderManager` to get the contacts, which gets them asyncly. Furthermore, I added a max of 50 results upon contacts loading, to get more specific result, the user can search the first name, last name, or number of the contacts.

Other improvements added was the use of the number formatter, which will make the results filter out numbers that cant exist. I also added the progress bar when contacts are loading.

**NOTE:**
There are 2 things I'm unsure of and would like some feed back
- Should I increase the number of results, or is 50 max results per query good?
- Is the progress bar useful. Maybe I didn't have enough contacts, but the loading was pretty fast for me. Tell me if it should be kept or removed if possible

### How to check PR:

1. Add many contacts to the Android device if you can
2. Run the app and login as any user if not already logged in
3. Click "Add Users"
4. Click "Invite from Contact List" (accept  permissions if requested)

You should see your contacts as the app always showed. 
Please also test the search bar.

### CODE REVIEW

File: app/src/main/java/com/example/bigowlapp/activity/SearchContactsToSupervise.java [hunk 1]
```diff
@@ -15,71 +15,92 @@
...
-        }
-        cursor.close();
 
-        listShow = list;
-        ArrayAdapter<String> adapter = new ArrayAdapter<>(getBaseContext(), android.R.layout.simple_list_item_1, listShow);
-        listContactsView = findViewById(R.id.listContacts);
-        listContactsView.setAdapter(adapter);
+            @Override
+            public void afterTextChanged(Editable editable) {
+                // Nothing happens aster user search input
```
Comment (Landry333) [inline app/src/main/java/com/example/bigowlapp/activity/SearchContactsToSupervise.java]: Probably just a typing error here, maybe you wanted to say "after"...

**Review by Landry333 — APPROVED** [review]
I tested the feature and it works fine as expected. As concerning your question, yes 50 users is a good number as the user will quickly realized that not all users from phone contacts were displayed in case of high number of contacts .You can leave the loading bar, it is appearing in case of high number of contacts like in my case. I only had a problem where the app doesn't go to the  SendingRequestToSuperviseActivity screen in case of existing user. But I guess as this wasn't part of your issue, we can check that problem in another issue. 
As for the code, it looks good to me so I will approve. Good job

File: app/src/main/java/com/example/bigowlapp/activity/SearchContactsToSupervise.java (hunk 1 above)
Comment (refatK) [reply app/src/main/java/com/example/bigowlapp/activity/SearchContactsToSupervise.java]: good catch, will fix

**Comment by refatK** [comment]
> I tested the feature and it works fine as expected. As concerning your question, yes 50 users is a good number as the user will quickly realized that not all users from phone contacts were displayed in case of high number of contacts .You can leave the loading bar, it is appearing in case of high number of contacts like in my case. I only had a problem where the app doesn't go to the SendingRequestToSuperviseActivity screen in case of existing user. But I guess as this wasn't part of your issue, we can check that problem in another issue.
> As for the code, it looks good to me so I will approve. Good job

Might have accidentally broke the existing number case. I'll check that.

**Comment by ShawnGuerra-Bautista** [comment]
I don't have this bug on my phone, but the feature works normally.

I checked the code; it's pretty clean. I see no problem; approved.

**Review by ShawnGuerra-Bautista — APPROVED** [review]

**Comment by Landry333** [comment]
I know this is a bit aside of what your did here, but could you also add a number formatter to the input phone number when user is searching contact by typing the contact phone number. You can make it as another BUG issue if you like

File: app/src/main/java/com/example/bigowlapp/activity/SearchContactsToSupervise.java (hunk 1 above)
Comment (refatK) [reply app/src/main/java/com/example/bigowlapp/activity/SearchContactsToSupervise.java]: @Landry333 Fixed

**Comment by refatK** [comment]
> I tested the feature and it works fine as expected. As concerning your question, yes 50 users is a good number as the user will quickly realized that not all users from phone contacts were displayed in case of high number of contacts .You can leave the loading bar, it is appearing in case of high number of contacts like in my case. I only had a problem where the app doesn't go to the SendingRequestToSuperviseActivity screen in case of existing user. But I guess as this wasn't part of your issue, we can check that problem in another issue.
> As for the code, it looks good to me so I will approve. Good job

@Landry333  You were right. I thought I fixed the issue but I had made a mistake. Please check again ad confirm there are no issues

**Comment by refatK** [comment]
> I know this is a bit aside of what your did here, but could you also add a number formatter to the input phone number when user is searching contact by typing the contact phone number. You can make it as another BUG issue if you like

did it

**Comment by Landry333** [comment]
I tested the fix you made. It is working fine, to make a request and also the number formatter to find a contact by typing a number. I know it was some kind of aside to the main thing you were doing, so good job!

**Review by ShawnGuerra-Bautista — APPROVED** [review]
Everything works as last time. Approved.

**Pull request merged**